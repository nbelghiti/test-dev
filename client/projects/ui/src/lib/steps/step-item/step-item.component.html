<div
	class="c-step-item"
	[ngClass]="{
		'c-step-item--completed': state === StepType.completed,
		'c-step-item--active': state === StepType.active
	}"
>
	<span class="c-step-item__number">
		{{ number }}
	</span>

	<div class="c-step-item__content">
		<p class="c-step-item__title">
			{{ title }}
		</p>

		<p class="c-step-item__description">
			{{ description }}
		</p>
	</div>
</div>
