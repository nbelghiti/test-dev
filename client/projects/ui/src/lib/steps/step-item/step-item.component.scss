// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Settings
// =============================================================================
$step-number-size: 6rem;

// =============================================================================
// :: Step Item
// =============================================================================
.c-step-item {
	display: flex;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-step-item__number {
	display: flex;
	justify-content: center;
	align-items: center;
	width: $step-number-size;
	height: $step-number-size;

	font-family: $font-family-tertiary;
	font-size: $font-size-4x-large;
	font-weight: $font-weight-extrablack;
	font-variant-numeric: tabular-nums lining-nums;
	line-height: $step-number-size;
	text-align: center;

	color: $text-color-grey-76;
	border-radius: 50%;
	background-color: $background-color-base;
}

.c-step-item__content {
	max-width: 17.5rem;
	margin-left: 1.5rem;
}

.c-step-item__title {
	font-size: $font-size-base;
	font-weight: $font-weight-bold;
	line-height: $line-height-base;

	color: $text-color-grey-56;
}

.c-step-item__description {
	font-size: $font-size-1x-small;
	line-height: $line-height-base;

	color: $text-color-grey-56;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-step-item--active {

	.c-step-item__number {
		color: $text-color-white;
		border: none;
		background-color: $primary-color;
	}

	.c-step-item__title {
		color: $text-color-base;
	}

	.c-step-item__description {
		color: $text-color-grey-46;
	}
}

.c-step-item--completed {

	.c-step-item__number {
		color: $text-color-white;
		border: none;
		background-color: $background-color-orange-78;
	}
}
