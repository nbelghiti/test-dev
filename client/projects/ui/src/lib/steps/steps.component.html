<nav class="c-steps">
	<ul class="c-steps__list">
		<li
			class="c-steps__item"
			[ngClass]="{'c-steps-item--completed' : i < activeIndex}"
			*ngFor="let step of steps; let i = index"
		>
			<ui-step-item
				[state]="activeIndex === i ? StepType.active : i < activeIndex ? StepType.completed : StepType.default"
				[number]="i + 1"
				[title]="step.title"
				[description]="step.description"
			></ui-step-item>
		</li>
	</ul>
</nav>
