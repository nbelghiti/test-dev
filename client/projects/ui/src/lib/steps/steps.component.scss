// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Settings
// =============================================================================
$step-number-size: 6rem;
$step-margin: 2rem;

// =============================================================================
// :: Steps
// =============================================================================
.c-steps {
	display: flex;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-steps__list {
	@include extend-list-clear();

	display: flex;
	flex-direction: column;
	align-items: flex-start;
}

.c-steps__item {
	position: relative;

	min-height: 8.5rem;
	margin: $step-margin 0;

	cursor: default;

	&:first-child {
		margin-top: 0;
	}

	&:last-child {
		margin-bottom: 0;

		&::after {
			content: none;
		}
	}

	&::after {
		content: "";

		position: absolute;
		left: calc(#{$step-number-size / 2});
		bottom: calc(#{-$step-number-size / 2});

		display: block;
		width: 0.4rem;
		height: calc(#{$step-margin * 2 + 0.5});

		background-color: $grey-86;

		transform: translateX(-50%);

		transition: background-color $animation-base;
	}
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-steps-item--completed {

	&::after {
		background-color: $background-color-orange-78;
	}
}
