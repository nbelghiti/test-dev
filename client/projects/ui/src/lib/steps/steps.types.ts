export enum StepType {
	default = '',
	active = 'active',
	completed  = 'completed',
}

export interface IStep {
	title: string;
	description: string;
}
