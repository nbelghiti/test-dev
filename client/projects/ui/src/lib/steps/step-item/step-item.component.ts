import { Component, Input } from '@angular/core';

import { StepType } from '../steps.types';

@Component({
	selector: 'ui-step-item',
	templateUrl: './step-item.component.html',
	styleUrls: ['./step-item.component.scss']
})

export class StepItemComponent {
	@Input() public number: number;
	@Input() public state: StepType = StepType.default;
	@Input() public title: string;
	@Input() public description: string;
	public StepType = StepType;
}
