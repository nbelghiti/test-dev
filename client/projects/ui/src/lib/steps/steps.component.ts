import { Component, Input } from '@angular/core';

import { IStep, StepType } from './steps.types';

@Component({
	selector: 'ui-steps',
	templateUrl: './steps.component.html',
	styleUrls: ['./steps.component.scss']
})

export class StepsComponent {
	@Input() public steps: IStep[];
	@Input() public activeIndex: number = 1;
	public StepType = StepType;
}
