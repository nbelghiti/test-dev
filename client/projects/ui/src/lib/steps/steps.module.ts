import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';

import { StepItemComponent } from './step-item/step-item.component';
import { StepsComponent } from './steps.component';

@NgModule({
	imports: [
		CommonModule,
		RouterModule
	],
	declarations: [
		StepItemComponent,
		StepsComponent,
	],
	exports: [
		StepItemComponent,
		StepsComponent,
	],
})
export class StepsModule {}
