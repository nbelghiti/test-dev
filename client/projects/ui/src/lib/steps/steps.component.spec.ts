import { ComponentFixture, TestBed, async } from '@angular/core/testing';

import { StepItemComponent } from './step-item/step-item.component';
import { StepsComponent } from './steps.component';

describe('StepsComponent', () => {
	let component: StepsComponent;
	let fixture: ComponentFixture<StepsComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			declarations: [
				StepsComponent,
				StepItemComponent,
			]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(StepsComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
