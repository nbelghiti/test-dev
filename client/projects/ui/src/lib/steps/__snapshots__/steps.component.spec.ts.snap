// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepsComponent Should render correctly 1`] = `
<ui-steps
  StepType={[Function Object]}
  activeIndex={[Function Number]}
>
  <nav
    class="c-steps"
  >
    <ul
      class="c-steps__list"
    >
      
    </ul>
  </nav>
</ui-steps>
`;
